flow/project.py

```diff
@@ -94,6 +96,21 @@
 {filters}"""
 
 
+class IgnoreConditions(IntEnum):
+    PRE = 1
+    POST = 2
+    ALL = PRE | POST
+    NONE = ~ ALL
+
+
+_IGNORE_CONDITIONS = {
+    'pre': IgnoreConditions.PRE,
+    'post': IgnoreConditions.POST,
+    'all': IgnoreConditions.ALL,
+    'none': IgnoreConditions.NONE
+}
```

Comment:
Following Simon's outstanding comment, we should be able to get rid of this entirely.